pkg/client/record/event.go

```diff
@@ -266,13 +276,22 @@ func (recorder *recorderImpl) generateEvent(object runtime.Object, timestamp unv
 	event := recorder.makeEvent(ref, eventtype, reason, message)
 	event.Source = recorder.source
 
+	recorder.wg.Add(2)
```

Comment:
adding 2 because we want to wait for the event is recorded in broadcaster but also processed using `recordToSink`